lib/autoBlock.js

```diff
@@ -29,48 +29,53 @@ var validBlockKeys = [
   'errorDefaults',
   'errorMapping',
   'func',
+  'name',
   'value',
   'when',
-  'with'
+  'with',
+  'withDefaults'
 ]
 
 // Runs a block using common block/controller pattern
 function runBlock (config, done) {
   config.data = config.data || {}
 
-  // Set config.done if not provided
-  if (!config.done) {
-    config.done = done || (() => {
-    })
+  var noop = () => {
   }
 
+  // Set config.done if not provided
+  config.done = config.done || done || noop
```

Comment:
Seems a little weird, what if both config.done and done are provided? It seems like done won't be called in that case.